### ISSUE TITLE
Keywords from Python libraries not recognized/marked as error

### ISSUE THREAD

**Issue by oliver-gramberg** [robocorp/robotframework-lsp#166]
**Describe the bug**
In the VS Code editor, keywords defined in Python files within the workspace are not recognized (are marked as error with a red squiggly underline) and/or cannot be jumped to using Ctrl+click/F12. On the other hand, running the tests works fine.

**To Reproduce**
In my workspace, I have `./tests/001_Login_Logout/001_login.robot` that contains, e.g.,
```
*** Test Cases ***
User can login in with valid credentials
    The current page should be    LoginPage
    Login             ${USERNAME}    ${PASSWORD}
```
VS Code shows the word `Login` with a red squiqqly underline.

I have `./env/Lib/site-packages/PageObjectLibrary/keywords.py`, and when I Ctrl-click `The current page should be` it takes me there.
I have `./lib/LoginPage.py` (next to an empty `./lib/__init__.py`) where `class LoginPage` resides. When I Ctrl-click `LoginPage` in the test, nothing happens.
That class has `def login(self, email, password):`. The keyword `Login` in the test has a red squiqqly underline, while `LoginPage` does not. When I Ctrl-click `Login` in the test, nothing happens.

settings.json:
```
{
  "workbench.colorTheme": "Default Light+",
  "python.pythonPath": "env\\Scripts\\python.exe",
  "terminal.integrated.shell.windows": "C:\\Windows\\System32\\cmd.exe",
  "robot.language-server.python": "C:\\<path to workspace>\\env\\Scripts\\python.exe",
  "robot.pythonpath": [
        "C:\\<path to workspace>\\lib",
      ],
}
```

**Expected behavior**
I'd like all keywords and things to be recognized in the editor and to be able to be looked up. (The whole project is taken from a Linux/macOS/IntelliJ environment where everything seems to work, we chose VS Code for its presumably superior debugging support.)

**Versions:**
 - OS: Windows 10
 - Robot Framework Version: 3.2.1
 - Robot Framework Language Server Version: 0.4.2
 - Client Version: VSCode 1.49.2

**Logs**
[temp.zip](https://github.com/robocorp/robotframework-lsp/files/5319323/temp.zip)

**Comment by fabioz** [comment]
I was able to reproduce this locally. I'm taking a look at why that happens.

**Comment by fabioz** [comment]
Actually, I wasn't able to reproduce it (I thought I had, but it was something unrelated on my side).

Now, taking a look at the logs, I see:

```
lsp: 2020-10-02 18:48:54 UTC pid: 23056 - ThreadPoolExecutor-1_0 - EXCEPTION - robotframework_ls.impl.libspec_manager
Error creating libspec: LoginPage. Output:
Initializing test library 'LoginPage' with no arguments failed: RobotNotRunningError: Cannot access execution context

Traceback (most recent call last):
  File "C:\Users\grambeol\AppData\Local\Programs\Python\Python38\lib\site-packages\PageObjectLibrary\pageobject.py", line 70, in __init__
    self._selib = SeleniumLibraryExecutor()
  File "C:\Users\grambeol\AppData\Local\Programs\Python\Python38\lib\site-packages\PageObjectLibrary\pageobject.py", line 20, in __init__
    self.selib = BuiltIn().get_library_instance("SeleniumLibrary")
  File "C:\Users\grambeol\AppData\Local\Programs\Python\Python38\lib\site-packages\robot\libraries\BuiltIn.py", line 3284, in get_library_instance
    return self._namespace.get_library_instance(name)
  File "C:\Users\grambeol\AppData\Local\Programs\Python\Python38\lib\site-packages\robot\libraries\BuiltIn.py", line 73, in _namespace
    return self._get_context().namespace
  File "C:\Users\grambeol\AppData\Local\Programs\Python\Python38\lib\site-packages\robot\libraries\BuiltIn.py", line 68, in _get_context
    raise RobotNotRunningError('Cannot access execution context')
```

So, the problem is that the libspec for the `LoginPage` cannot be created because importing it just to create the libspec fails. 

The fix here is making it possible to instance that class without having the actual execution environment (maybe handling the `RobotNotRunningError` or making `selib` a property that's lazily initialized -- although that'd only work if you don't do anything else with it in the constructor).

So, I'm closing this issue because I don't think there's much that can be done in the extension (if it can't introspect your library it's not able to get proper information for code analysis), but feel free to comment more in the issue.

**Comment by fabioz** [comment]
Note: the command line being used to generate the libspec is:

```C:\Users\grambeol\AppData\Local\Programs\Python\Python38\python.exe -m robot.libdoc --format XML:HTML -P c:\Users\grambeol\Smplcty\smplcty-integration-test\tests\000_global_resources -P C:\Users\grambeol\Smplcty\smplcty-integration-test\lib LoginPage C:\Users\grambeol\.robotframework-ls\specs\6ef74466_3.2.1\user\LoginPage.libspec```

So, you can test if your changes to the structure you have worked by executing that in the command line.

**Comment by fabioz** [comment]
Note: there's another related error: `Initializing test library 'DeviceListPage' with no arguments failed: RobotNotRunningError: Cannot access execution context`, so, that's another library that also won't work due to that.

**Comment by oliver-gramberg** [comment]
Thanks for your help so far. Making selib a property that's lazily initialized actually does the trick.

The fix would have to be done in https://github.com/boakley/robotframework-pageobjectlibrary/blob/master/PageObjectLibrary/pageobject.py, and when I looked there to suggest a fix, I found that I had an older version from a cache. The current version does the same, so the solution is to 'pip install --update'...
